src/transformers/models/encoder_decoder/modeling_flax_encoder_decoder.py

```diff
@@ -0,0 +1,819 @@
+
+            >>> tokenizer = BertTokenizer.from_pretrained('bert-base-cased')
+
+            >>> # forward
+            >>> input_ids = tokenizer.encode("Hello, my dog is cute", add_special_tokens=True, return_tensors='np')  # Batch size 1
+            >>> outputs = model(input_ids=input_ids, decoder_input_ids=input_ids)
+
+            >>> # save and load from pretrained
+            >>> model.save_pretrained("bert2gpt2")
+            >>> model = FlaxEncoderDecoderModel.from_pretrained("bert2gpt2")
+
+            >>> # generation (use GPT2's `bos_token_id` to start the generation.)
+            >>> eos_token_id = model.config.decoder.eos_token_id
+            >>> pad_token_id = eos_token_id
+            >>> decoder_start_token_id = model.config.decoder.bos_token_id
+            >>> generated = model.generate(input_ids, pad_token_id=pad_token_id, eos_token_id=eos_token_id, decoder_start_token_id=decoder_start_token_id)
+
+        """
+        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+        return_dict = return_dict if return_dict is not None else self.config.return_dict
+
+        # prepare encoder inputs
+        if attention_mask is None:
+            attention_mask = jnp.ones_like(input_ids)
+        if position_ids is None:
+            batch_size, sequence_length = input_ids.shape
+            position_ids = jnp.broadcast_to(jnp.arange(sequence_length)[None, :], (batch_size, sequence_length))
+
+        if decoder_attention_mask is None:
+            decoder_attention_mask = jnp.ones_like(decoder_input_ids)
+        if decoder_position_ids is None:
+            batch_size, sequence_length = decoder_input_ids.shape
+            decoder_position_ids = jnp.broadcast_to(
+                jnp.arange(sequence_length)[None, :], (batch_size, sequence_length)
+            )
+
+        # Handle any PRNG if needed
+        rngs = {"dropout": dropout_rng} if dropout_rng is not None else {}
+
+        return self.module.apply(
+            {"params": params or self.params},
+            input_ids=jnp.array(input_ids, dtype="i4"),
+            attention_mask=jnp.array(attention_mask, dtype="i4"),
+            decoder_input_ids=jnp.array(decoder_input_ids, dtype="i4"),
+            decoder_attention_mask=jnp.array(decoder_attention_mask, dtype="i4"),
+            position_ids=jnp.array(position_ids, dtype="i4"),
+            decoder_position_ids=jnp.array(decoder_position_ids, dtype="i4"),
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+            deterministic=not train,
+            rngs=rngs,
+        )
+
+    def prepare_inputs_for_generation(
```

Comment:
@patil-suraj In PyTorch we actually shift the reponsability of creating the model inputs to the decoder class, *e.g.* we call: 

```
self.decoder.prepare_inputs_for_generation(...)
```

In Flax this is more difficult since we can't do `self.decoder.prepare_inputs_for_generation` as its a module and not a model. I'm very much fine with the current solution as it covers all the relevant cases IMO (position_ids & attention_mask).

So I'm happy to go forward with it - let me know if you disagree here